src/main/kotlin/com/github/mitchellolsthoorn/testgenie/evosuite/EvoSuiteResultWatcher.kt

```diff
@@ -0,0 +1,54 @@
+package com.github.mitchellolsthoorn.testgenie.evosuite
+
+import com.google.gson.Gson
+import com.google.gson.stream.JsonReader
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.io.FileUtilRt
+import org.evosuite.utils.CompactReport
+import java.io.File
+import java.io.FileReader
+
+
+class EvoSuiteResultWatcher(private val project: Project, private val resultPath: String) : Runnable {
+    private val log = Logger.getInstance(EvoSuiteResultWatcher::class.java)
+
+    override fun run() {
+        val sep = File.separatorChar
+        val testResultDirectory = "${FileUtilRt.getTempDirectory()}${sep}testGenieResults${sep}"
```

Comment:
I admit, this has been done very smartly

Reply:
Actually, I find it pretty terrible and I'd want to isolate this chunk of code as soon as possible.